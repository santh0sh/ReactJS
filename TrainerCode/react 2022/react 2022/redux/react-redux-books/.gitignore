node_modules
npm-debug.log
.npmrc
bin
.idea
*.iml
