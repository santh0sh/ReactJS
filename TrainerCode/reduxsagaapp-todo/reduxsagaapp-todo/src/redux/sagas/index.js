import { spawn } from 'redux-saga/effects'

// Sagas
import todoSaga from './todo-saga'

// Export the root saga
export default function* rootSaga() {
	console.log("Working with  Redux-Saga!")
	
	yield spawn(todoSaga)
}