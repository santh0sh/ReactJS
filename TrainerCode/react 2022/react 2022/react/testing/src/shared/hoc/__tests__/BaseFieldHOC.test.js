import renderer from 'react-test-renderer';
import BaseFieldHOC from '../BaseFieldHOC';

import { reduxForm } from 'redux-form';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

describe('Render BaseFieldHOC', () => {
    const store = createStore(() => ({}));
    let BaseFieldHOCComponent;

    beforeEach(() => {
        const TextInput = () => { return 'text input'; },
            BaseFieldHOCWrapper = BaseFieldHOC(TextInput),
            TextField = reduxForm({ form: 'testForm' })(BaseFieldHOCWrapper);
        BaseFieldHOCComponent = renderer.create(
            <Provider store={store}>
                <TextField name="text-input" />
            </Provider>
        ).toJSON();
    });

    it('render correctly component', () => {
        expect(BaseFieldHOCComponent).toMatchSnapshot();
    });

    it('check input component is wrapped in BaseFieldLayout', () => {
        expect(BaseFieldHOCComponent.props.className).toEqual('form-group');
    });
});
