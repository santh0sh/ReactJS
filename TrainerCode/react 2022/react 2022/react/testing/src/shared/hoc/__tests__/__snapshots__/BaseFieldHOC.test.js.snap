// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BaseFieldHOC render correctly component 1`] = `
<div
  className="form-group"
>
  <label
    className=" control-label"
    htmlFor="text-input"
  />
  <div
    className="control-field"
  >
    <div
      className="control-element"
    >
      text input
    </div>
  </div>
</div>
`;
